epochStart/metachain/validators.go

```diff
@@ -179,43 +188,62 @@ func (r *validatorInfoCreator) VerifyValidatorInfoMiniBlocks(
 }
 
 // SaveValidatorInfoBlocksToStorage saves created data to storage
-func (r *validatorInfoCreator) SaveValidatorInfoBlocksToStorage(metaBlock *block.MetaBlock, body *block.Body) {
+func (vic *validatorInfoCreator) SaveValidatorInfoBlocksToStorage(_ *block.MetaBlock, body *block.Body) {
+	if check.IfNil(body) {
+		return
+	}
+
 	for _, miniBlock := range body.MiniBlocks {
 		if miniBlock.Type != block.PeerBlock {
 			continue
 		}
 
-		for _, mbHeader := range metaBlock.MiniBlockHeaders {
-			if mbHeader.Type != block.PeerBlock {
-				continue
-			}
-
-			marshaledData, err := r.marshalizer.Marshal(miniBlock)
-			if err != nil {
-				continue
-			}
-
-			mbHash := r.hasher.Compute(string(marshaledData))
-			if !bytes.Equal(mbHeader.Hash, mbHash) {
-				continue
-			}
-
-			_ = r.miniBlockStorage.Put(mbHeader.Hash, marshaledData)
-			break
+		marshalizedData, err := vic.marshalizer.Marshal(miniBlock)
+		if err != nil {
+			continue
 		}
+
+		mbHash := vic.hasher.Compute(string(marshalizedData))
+		_ = vic.miniBlockStorage.Put(mbHash, marshalizedData)
 	}
 }
 
 // DeleteValidatorInfoBlocksFromStorage deletes data from storage
-func (r *validatorInfoCreator) DeleteValidatorInfoBlocksFromStorage(metaBlock *block.MetaBlock) {
+func (vic *validatorInfoCreator) DeleteValidatorInfoBlocksFromStorage(metaBlock *block.MetaBlock) {
+	if check.IfNil(metaBlock) {
+		return
+	}
+
 	for _, mbHeader := range metaBlock.MiniBlockHeaders {
 		if mbHeader.Type == block.PeerBlock {
-			_ = r.miniBlockStorage.Remove(mbHeader.Hash)
+			_ = vic.miniBlockStorage.Remove(mbHeader.Hash)
 		}
 	}
 }
 
 // IsInterfaceNil return true if underlying object is nil
-func (r *validatorInfoCreator) IsInterfaceNil() bool {
-	return r == nil
+func (vic *validatorInfoCreator) IsInterfaceNil() bool {
+	return vic == nil
+}
+
+// RemoveBlockDataFromPools removes block info from pools
+func (vic *validatorInfoCreator) RemoveBlockDataFromPools(metaBlock *block.MetaBlock, _ *block.Body) {
+	if check.IfNil(metaBlock) {
+		return
+	}
+
+	for _, mbHeader := range metaBlock.MiniBlockHeaders {
+		if mbHeader.Type != block.PeerBlock {
+			continue
+		}
+
+		vic.dataPool.MiniBlocks().Remove(mbHeader.Hash)
```

Comment:
this type of multiple inclusions is not go idiomatic - should save vic.dataPool.MiniBlocks into a variable - same as for transactionPool.

Reply:
Done